### PR TITLE
Update delfin driver developer guide for new resources and APIs

### PR DESCRIPTION
**Pull request by joseph-v**
**What type of PR is this?**

> /kind documentation

**What this PR does / why we need it**:
Update delfin driver developer guide for new resources and APIs

**Which issue(s) this PR fixes**:

Fixes #

**Test Report Added?**:
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
> /kind TESTED
> /kind NOT-TESTED

**Test Report**:

**Special notes for your reviewer**:

### CODE REVIEW

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md [hunk 1]
```diff
@@ -83,19 +83,28 @@ Existing Delfin Drivers for reference:
...
 | **remove_trap_config()** | `context`: Delfin framework context, `trap_config`: trap configuration | None |  Remove trap receiver configuration from storage system |
 | **parse_alert()** | `context`: Delfin framework context, `alert`: alert to parse | Alert dict as shown in `delfin/drivers/driver.py` |  Parse alert data got from snmp trap server |
 | **list_alerts()** | `context`: Delfin framework context, `query_para`: optional, contains 'begin_time' and 'end_time' | Alert dict as shown in `delfin/drivers/driver.py` |  List all current alerts from storage system |
 | **clear_alert()** | `context`: Delfin framework context, `alert`: alert to clear | Success/Failure |   Clear alert from storage system |
+| **get_capabilities()** | `context`: Delfin framework context | Capabilities schema defined in `delfin/api/schemas/storage_capabilities_schema.py` | Return metrics collection capabilites of the storage system |
+| **collect_perf_metrics()** | `context`: Delfin framework context, `storage_id`: delfin storage id, `resource_metrics`: required metrics, `start_time`: metrics start time, `end_time`: metrics end time | Collected metrics |   Collects metrics from storage system |
```
Comment (NajmudheenCT) [inline content/guides/developer-guides/delfin/driver-developer-guide/_index.md:107]: Give a sample response example 
```
# A list of delfin.common.costants.metric_struct 
 [
   "name=""throughput",
   "labels="{
      "storage_id":"7c0aa42c-d744-4353-84da-3c9b7f631f5a",
      "resource_type":"storage",
      "resource_id":"storage_0",
      "type":"RAW",
      "unit":"MB/s",
      "name":"fake_driver",
      "serial_number":"c83228e5-77d8-40c5-aa4b-2d6067d73c8b"
   },
   "values="{
      "1624874092000":75.63737435785605
   }
]
```

**Comment by sushanthakumar** [comment]
LGTM

File: content/guides/developer-guides/delfin/driver-developer-guide/_index.md (hunk 1 above)
Comment (joseph-v) [reply content/guides/developer-guides/delfin/driver-developer-guide/_index.md:107]: Updated

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Review by kumarashit — APPROVED** [review]
LGTM

**Pull request merged**